subprojects/platform-play/src/main/java/org/gradle/play/internal/toolchain/PlayToolChainServiceRegistry.java

```diff
@@ -38,8 +39,8 @@ public void registerProjectServices(ServiceRegistration registration) {
     }
 
     private static class ProjectScopeCompileServices {
-        PlayToolChainInternal createPlayToolChain(FileResolver fileResolver, WorkerDaemonFactory workerDaemonFactory, ConfigurationContainer configurationContainer, DependencyHandler dependencyHandler, WorkerProcessFactory workerProcessBuilderFactory) {
-            return new DefaultPlayToolChain(fileResolver, workerDaemonFactory, configurationContainer, dependencyHandler, workerProcessBuilderFactory);
+        PlayToolChainInternal createPlayToolChain(FileResolver fileResolver, WorkerDaemonFactory workerDaemonFactory, WorkerDirectoryProvider workerDirectoryProvider, ConfigurationContainer configurationContainer, DependencyHandler dependencyHandler, WorkerProcessFactory workerProcessBuilderFactory) {
+            return new DefaultPlayToolChain(fileResolver, workerDaemonFactory, workerDirectoryProvider, configurationContainer, dependencyHandler, workerProcessBuilderFactory);
```

Comment:
Yeah, by changing the idle directory to be injected on the server side, this extra service is no longer needed (which cleans this up a good bit).